examples/src/app/code-editor.mjs

```diff
@@ -59,27 +65,76 @@ class CodeEditor extends TypedComponent {
 
     componentDidMount() {
         this.handleExampleLoad = this.handleExampleLoad.bind(this);
+        this.handleExampleError = this.handleExampleError.bind(this);
         this.handleExampleLoading = this.handleExampleLoading.bind(this);
+        this.handleExampleHotReload = this.handleExampleHotReload.bind(this);
         this.handleRequestedFiles = this.handleRequestedFiles.bind(this);
         window.addEventListener('exampleLoad', this.handleExampleLoad);
+        window.addEventListener('exampleError', this.handleExampleError);
         window.addEventListener('exampleLoading', this.handleExampleLoading);
+        window.addEventListener('exampleHotReload', this.handleExampleHotReload);
         window.addEventListener("requestedFiles", this.handleRequestedFiles);
         iframeRequestFiles();
     }
 
     componentWillUnmount() {
         window.removeEventListener("exampleLoad", this.handleExampleLoad);
+        window.removeEventListener("exampleError", this.handleExampleError);
         window.removeEventListener("exampleLoading", this.handleExampleLoading);
+        window.removeEventListener("exampleHotReload", this.handleExampleHotReload);
         window.removeEventListener("requestedFiles", this.handleRequestedFiles);
     }
 
+    handleExampleHotReload() {
+        this.clearDecorations();
+        // console.log("handleExampleHotReload> clear decorations");
+    }
+
     handleExampleLoad(event) {
         // console.log("CodeEditor got files event", event);
         /** @type {Record<string, string>} */
         const files = event.files;
         this.mergeState({ files, selectedFile: 'example.mjs' });
     }
 
+    /**
+     * @param {CustomEvent} event - Custom event containing error thrown in iframe.
+     */
+    handleExampleError(event) {
+        const { editor } = this;
+        if (!editor) {
+            return;
+        }
+        // error can be an actual error or e.g. only a string in case of throw "oh ohhh"
+        const error = event.detail;
+        if (!error.message) {
+            return;
+        }
+        const message = `**Error**\n${error.message}\n\n**Stack**\n ${error.stack}`;
+        const lines = error.stack.split('\n');
+        const firstLineWithAt = lines.find(line => / *at/.test(line));
+        // -2 to map to correct line in editor
+        const line = firstLineWithAt.split(':').at(-2) - 2;
+        const lineText = editor.getModel().getLineContent(line);
```

Comment:
Nice work @kungfooman 🙏 I may be wrong, but is it better to use `setModelMarkers` here?  https://microsoft.github.io/monaco-editor/playground.html?source=v0.44.0#example-extending-language-services-model-markers-example. Not 100% clear on the difference but seems it's more generally used for errors

Reply:
Interesting, thank you @marklundin... that API even looks a bit simpler, just `monaco.editor.removeAllMarkers()` instead of passing around "delta arrays".